src/core/SampleBuffer.cpp

```diff
@@ -1416,7 +1416,7 @@ QString SampleBuffer::tryToMakeRelative( const QString & file )
 	if( QFileInfo( file ).isRelative() == false )
 	{
 		// Normalize the path
-		QString f = QFileInfo( file ).canonicalFilePath().replace( QDir::separator(), '/' );
+		QString f( QDir::cleanPath( file ).replace( QDir::separator(), '/' ) );
```

Comment:
Just a reminder, `.replace( QDir::separator(), '/' )` can be removed when Qt4 support is completely dropped later.

Reply:
> Just a reminder, `.replace( QDir::separator(), '/' )` can be removed when Qt4 support is completely dropped later.

Good to know... can you elaborate?  Is the default behavior to use `/` in Qt5?  If so where is this documented and is this across all Qt functions?

Reply:
Documented in http://doc.qt.io/qt-5/qdir.html#cleanPath

Reply:
> Documented in http://doc.qt.io/qt-5/qdir.html#cleanPath

Yeah that doesn't have any "since qt5" warnings, so I just thought it was an example.  Thanks for clarification.

Reply:
For comparison:

4.8:

> *Removes all multiple directory separators `"/"` and resolves any `"."`s or `".."`s found in the path, path.*

5.x:

> *Returns path with directory separators normalized (converted to `"/"`) and redundant ones removed, and `"."`s and `".."`s resolved (as far as possible).*

So I hope you understand my hesitation. :)

Reply:
Hmm.... Here's the code for posterity... 

```cpp

static QString qt_cleanPath(const QString &path, bool *ok)
--
{
if (path.isEmpty())
return path;
QString name = path;
QChar dir_separator = QDir::separator();
if (dir_separator != QLatin1Char('/'))
name.replace(dir_separator, QLatin1Char('/'));
 
QString ret = qt_normalizePathSegments(name, OSSupportsUncPaths, ok);
 
// Strip away last slash except for root directories
if (ret.length() > 1 && ret.endsWith(QLatin1Char('/'))) {
#if defined (Q_OS_WIN)
#  if defined(Q_OS_WINRT)
if (!((ret.length() == 3 \|\| ret.length() == QDir::rootPath().length()) && ret.at(1) == QLatin1Char(':')))
#  else
if (!(ret.length() == 3 && ret.at(1) == QLatin1Char(':')))
#  endif
#endif
ret.chop(1);
}
 
return ret;
}


```

The only function that seems like it would touch `\\` would be the `qt_normalizePathSegments.cpp`, but that specifically says in the source

```cpp
// This method is shared with QUrl, so it doesn't deal with QDir::separator() ...
```

I guess I'll take your word for it. :)

Reply:
@tresf it's about a third of the way down in that `qt_cleanPath` function:
```
name.replace(dir_separator, QLatin1Char('/'));
```

How do you feel about adding a short code comment explaining why we run `.replace( QDir::separator(), '/' )` (maybe just linking to https://github.com/LMMS/lmms/pull/4271 ) so that nobody else accidentally removes it, forgetting about Qt4 support?

Reply:
Looking into Qt4's code, I found that even Qt 4.0 has the replacement logic. So we can safely remove that.

Reply:
`dir_separator`, got it!  Missed that one.  Thanks!

> So we can safely remove that.

Agreed.  Will do.